packages/repo/tests/sync/diff.test.ts

```diff
@@ -0,0 +1,72 @@
+import * as crypto from '@atproto/crypto'
+import { DidResolver } from '@atproto/did-resolver'
+import { Repo, RepoContents } from '../../src'
+import { MemoryBlockstore } from '../../src/storage'
+import * as sync from '../../src/sync'
+
+import * as util from '../_util'
+
+describe('Diff Sync', () => {
+  let storage: MemoryBlockstore
+  let syncStorage: MemoryBlockstore
+  let repo: Repo
+  let keypair: crypto.Keypair
+  let repoData: RepoContents
+  const didResolver = new DidResolver()
+
+  beforeAll(async () => {
+    storage = new MemoryBlockstore()
+    keypair = await crypto.Secp256k1Keypair.create()
+    repo = await Repo.create(storage, keypair.did(), keypair)
+    syncStorage = new MemoryBlockstore()
+  })
+
+  let syncRepo: Repo
+
+  it('syncs an empty repo', async () => {
+    const car = await sync.getFullRepo(storage, repo.cid)
+    const loaded = await sync.loadFullRepo(syncStorage, car, didResolver)
+    syncRepo = await Repo.load(syncStorage, loaded.root)
+    const data = await syncRepo.data.list(10)
+    expect(data.length).toBe(0)
+  })
+
+  it('syncs a repo that is starting from scratch', async () => {
+    const filled = await util.fillRepo(repo, keypair, 100)
+    repo = filled.repo
+    repoData = filled.data
+
+    const car = await sync.getFullRepo(storage, repo.cid)
+    const loaded = await sync.loadFullRepo(syncStorage, car, didResolver)
+    syncRepo = await Repo.load(syncStorage, loaded.root)
+    const contents = await syncRepo.getContents()
+    expect(contents).toEqual(repoData)
+    await util.verifyRepoDiff(loaded.writeLog, {}, repoData)
+  })
+
+  it('syncs a repo that is behind', async () => {
+    // add more to providers's repo & have consumer catch up
+    const beforeData = JSON.parse(JSON.stringify(repoData))
```

Comment:
There's a nice way to do this now:
```suggestion
    const beforeData = structuredClone(repoData)
```

Reply:
oh beautiful how did i not know this